src/main/java/org/jabref/gui/search/GlobalSearchResultDialog.java

```diff
@@ -35,14 +37,15 @@ public class GlobalSearchResultDialog extends BaseDialog<Void> {
     private final UndoManager undoManager;
     private final LibraryTabContainer libraryTabContainer;
 
+    // Reference needs to be kept, since java garbage collection would otherwise destroy the subscription
```

Comment:
I don't understand this comment; why would we want to keep a reference to the subscription if we are not going to use it?

The actual listener would be attached to `viewModel.keepOnTop` so as long as we have a reference to that, we are safe. AFAIK subscriptions are only kept when we want to detach the listener later to avoid memory leaks.

Reply:
Yeah, sounds great in theory. But not in fact. Lot of painful iavafx experience here.